sys/include/crypto/chacha20poly1305.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2018 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_crypto_chacha20poly1305 chacha20poly1305 AEAD cipher
+ * @ingroup     sys_crypto
+ * @brief       Provides RFC 8439 style chacha20poly1305
+ *
+ * This module provides the chacha20poly1305 AEAD symmetric cipher following
+ * [rfc 8439](https://tools.ietf.org/html/rfc8439).
+ *
+ * Nonces must be unique per message for a single key. They are allowed to be
+ * predictable, e.g. a message counter and are allowed to be visible during
+ * transmission.
+ * @{
+ *
+ * @file
+ * @brief       Chacha20poly1305 functions
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef CRYPTO_CHACHA20POLY1305_H
+#define CRYPTO_CHACHA20POLY1305_H
+
+#include "crypto/poly1305.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CHACHA20POLY1305_KEY_BYTES      (32U) /**< Key length in bytes */
+#define CHACHA20POLY1305_NONCE_BYTES    (12U) /**< Nonce length in bytes */
+#define CHACHA20POLY1305_TAG_BYTES      (16U) /**< Tag length in bytes */
+
+/**
+ * @brief Chacha20poly1305 state struct
+ */
+typedef struct
```

Comment:
hmm I forgot about this...
Maybe change it to:
```c
typedef union
{
    /* We need both the state matrix and the poly1305 state, but nearly not at
     * the same time. This works as long as the first 8 members of state
     * overlap fully or completely not with the first and second key parts
     * from the @ref poly1305_ctx_t struct */

    uint32_t state[16];  /**< The current state of the key stream. */
    poly1305_ctx_t poly; /**< Poly1305 state for the MAC */

} chacha20poly1305_ctx_t;
```
To make it c99 compliant.

Reply:
Fixed, thanks!